### ISSUE TITLE
cannot reshape array of size  into shape

### ISSUE THREAD

**Issue by QLeelulu** [SerpentAI/SerpentAI#62]
MacOS 10.12.6, MacBook Pro (13-inch, 2017 with touch bar).

The SerpentAI version is:

```
pip install SerpentAI==0.1.6b1
```

```
 % serpent play BosonX SerpentBosonXAgentGameAgent
{'width': 1137, 'height': 642, 'x_offset': 0, 'y_offset': 43}
Traceback (most recent call last):
  File "/Users/qleelulu/dev/work/pyenv/SerpentAI/bin/serpent", line 11, in <module>
    sys.exit(execute())
  File "/Users/qleelulu/dev/work/pyenv/SerpentAI/lib/python3.6/site-packages/serpent/serpent.py", line 48, in execute
    command_function_mapping[command](*sys.argv[2:])
  File "/Users/qleelulu/dev/work/pyenv/SerpentAI/lib/python3.6/site-packages/serpent/serpent.py", line 237, in play
    game.play(game_agent_class_name=game_agent_name, frame_handler=frame_handler)
  File "/Users/qleelulu/dev/work/pyenv/SerpentAI/lib/python3.6/site-packages/serpent/game.py", line 157, in play
    game_frame = self.grab_latest_frame()
  File "/Users/qleelulu/dev/work/pyenv/SerpentAI/lib/python3.6/site-packages/serpent/game.py", line 215, in grab_latest_frame
    3
  File "/Users/qleelulu/dev/work/pyenv/SerpentAI/lib/python3.6/site-packages/serpent/frame_grabber.py", line 102, in get_frames
    frame_array = np.fromstring(frame_bytes, dtype="uint8").reshape(frame_shape)
ValueError: cannot reshape array of size 2218752 into shape (642,1137,3)
```

And #30 #31 don't work for me.

**Comment by nbrochu** [comment]
This is a known issue on macOS, one which I can't reproduce on my 10.12.6 MBP. The only logical thing I can see is it's a retina display rescale issue. If you go to System Preferences => Displays, are you on *Default for display* resolution?

I'm going to be looking into this one today and try to get a fix in 0.1.7b1 because it's one of the most widespread issues so far.

**Comment by QLeelulu** [comment]
Yes, i am on **Default for display** resolution. 
My MBP is retina display.

**Comment by nbrochu** [comment]
OK making this issue the main one for this bug.

**Comment by nbrochu** [comment]
Perfect! I can finally reproduce this issue with BosonX on my MPB. I get the exact same stacktrace. Looking into it.

**Comment by nbrochu** [comment]
The fix is in! It will be included in 0.1.7b1 which I will release later today.

I finally noticed that on macOS with retina displays, this only happened with not even-valued widths or heights in the game resolution. Fix is to subtract 1 from the odd values and resize the window.

https://github.com/SerpentAI/SerpentAI/commit/8543019b062d169a69aa34587e06a62575017d0c
